tests/testthat/test-board_azure_adls2.R

```diff
@@ -0,0 +1,14 @@
+board_adls <- board_azure_test(path = "", type = "dfs")
```

Comment:
Can you please put all the tests together in `test-board_azure.R`? Our convention is to match the test file name to the `R/` file name

Reply:
I put them in different files because it was actually easier this way: I can run tests on one service, without having to repeat all the tests for all the other services. Since the tests can take minutes to run, this is way faster.

I would suggest breaking your convention just this once. 😉

Reply:
If they take minutes to run the tests, then they probably need to be handled differently.

Reply:
Ok, I misremembered. I ran the tests again, and each file (azure_adls2, azure_blob, azure_file) takes about 30 seconds to run.

I'm actually running your standard test suite twice for each service: once for a board in the root directory, and again for a board in a supplied path. Testing both scenarios has actually been useful, there are some corner cases in how each service handles paths that I had to nail down.

Waiting 30 seconds for the test to conclude is better than waiting 1.5 minutes, so I still think it's worthwhile to separate them out. Unless there's a way to run only a subset of the tests in a file?

Reply:
Oh, one more thing: the SAS generated by the Azure portal won't work for the ADLS2 tests because it uses an outdated API version. You can use the AzureStor package to generate it instead.

```r
library(AzureStor)
endp <- storage_endpoint("https://xxxx.dfs.core.windows.net", key="yyyy")
sas <- get_account_sas(endp, permissions="rwcdl")
Sys.setenv(PINS_AZURE_SAS=sas)
```

Should I add a comment to this effect somewhere?

Reply:
Ok, leave them as is for now; I think I'll probably set up something custom to skip these tests for me when I'm developing locally and not touching this specific code (i.e. something with `skip_if` and an env var)

Reply:
Right now the tests should be skipped if all the necessary env vars aren't set. The way I usually do it is to have a batch script that sets these vars; if I want to skip tests that require auth, I just don't run the script beforehand. Your workflow may be different though.